File: 219-contains-duplicate-ii/219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        di=dict()
        for i in nums:
            di[i]=[]
        for i in range(len(nums)):
            di[nums[i]].append(i)
        for key,value in di.items():
            for m in range(1,len(value)):
                if value[m]-value[m-1]<=k:
                    return True
        return False

File: 219-contains-duplicate-ii/219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        map<int,vector<int>> mp;
        for(int i=0;i<nums.size();++i)
                mp[nums[i]].push_back(i);
        for(auto i:mp){
            auto tmp=i.second;
            for(int j=1;j<tmp.size();++j){
                if(tmp[j]-tmp[j-1]<=k)
                    return true;
            }
        }
        return false;
    }
};